lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -226,9 +230,12 @@ def paths_to_3d_segments_with_codes(paths, zs=0, zdir='z'):
 
     path_generator = (path_to_3d_segment_with_codes(path, pathz, zdir)
                       for path, pathz in zip(paths, zs))
-    segments, codes_list = zip(*path_generator)
+    if len(paths) and len(zs):
+        segments, codes_list = zip(*path_generator)
 
-    return np.asarray(segments), np.asarray(codes_list)
+        return np.asarray(segments), np.asarray(codes_list)
+    else:
+        return np.empty((3, 0)), np.array([])
```

Comment:
This should probably have explicit dtypes. What type are `segments` and `codes_list` normally?